fundrebalancer/fundrebalancer.go

```diff
@@ -674,9 +675,50 @@ func (r *FundRebalancer) isGasAcceptable(ctx context.Context, txns []SkipGoTxnWi
 		return false, "", fmt.Errorf("parsing max gas cost threshold")
 	}
 
-	if gasCostUUSDC.Cmp(maxCost) > 0 {
+	if gasCostUUSDC.Cmp(maxCost) <= 0 {
+		return true, gasCostUUSDC.String(), nil
+	}
+
+	// No fund rebalancing timeout set
+	if *chainConfig.FundRebalancingTimeout == -1 {
+		return false, gasCostUUSDC.String(), nil
+	}
+
+	transfers, err := r.database.GetPendingRebalanceTransfersToChain(ctx, chainID)
+	if err != nil {
+		return false, "", fmt.Errorf("getting pending transfers: %w", err)
+	}
+
+	if len(transfers) == 0 || chainConfig.FundRebalancingTimeout == nil {
```

Comment:
you are checking if `chainConfig.FundRebalancingTimeout` is nil here, but above you have already dereferenced it, which implies that the dereference on line 683 could be a nil pointer dereference